### ISSUE TITLE
Provide a "force language" switch

### ISSUE THREAD

**Issue by glatzert** [MicrosoftDocs/Docs#58]
Since my OS is german and my browser is Edge, I cannot adjust my preferred languages in the browser without affecting the OS.
Using Google or Bing to find documentation (e.g. search for T-SQL UPDATE FROM) will return a link in the form "https://docs.microsoft.com/de-de/sql/t-sql/queries/update-transact-sql" which is translated automatically - and to be honest - is too unprecise to read.
So I end up switching to en-us all the time, which is unneccessary and annoying.

I'd like to ask you, if it would be possible to provide a language switch, that sets a non-expiring Cookie and will take preference over the url-provided language.

**Comment by mairaw** [comment]
@adkinn @DuncanmaMSFT to look at this request.

**Comment by DuncanmaMSFT** [comment]
@glatzert when you pick a language from the bottom locale selector (in the footer of the site), are you finding that is going away too quickly? I'm checking the code right now and we *intend* for it to be set for a long time. Adding @jdanyow too as I think he wrote the locale picker code

**Comment by glatzert** [comment]
Hey @DuncanmaMSFT, it's not about "going away too quickly". What I was suggesting is a force, which allows me to set a language "for good".

So I visited https://docs.microsoft.com and actively set the language to en-US - thus far no problem.
Then I went over to google and searched "MS Docs ToString" (without quotes, just for the example)
https://www.google.de/search?q=MS+Docs+toString&ie=&oe= which will show this: https://docs.microsoft.com/de-de/dotnet/csharp/programming-guide/classes-and-structs/how-to-override-the-tostring-method

That search result contains my browser language (probably google preferres that, since I'm german).
Now if I click on that link, I will have to reset the language to en-us, if I want to read the docs in english (which does apply like 99.99% of the time). And because of that, I proposed a force language switch, which will always prefer the cookie over the url-language.

**Comment by aldonetti** [comment]
@glatzert it indeed sounds like Google is finding a German article and that's what you get when you click on that url. I believe you can force Google to only find articles in English if that's what you want.  For Duncan, I guess the change would be to have the cookie take precedence over the locale in the url but various scenarios need to be carefully thought through as trying to outsmart user intentions didn't always get the intended results in msdn...

**Comment by jdanyow** [comment]
@aldonetti is correct- if a link has the locale in it (`de-de` in this case), docs will not redirect to the user's preferred locale (stored in the `MarketplaceSelectedLocale` cookie).

If you take the locale out of the url, docs will redirect to your preferred locale. Example:

https://docs.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/how-to-override-the-tostring-method

**Comment by jdanyow** [comment]
@DuncanmaMSFT the locale cookie expires after 10 years :older_man: :hourglass_flowing_sand:

**Comment by glatzert** [comment]
@aldonetti since I don't want to mess with my Google, I proposed to provide a switch to let the user choose their intentions. Not outsmarting anyone - just allowing to force nice docs over machine-translated ones (which are indeed pretty funny to read sometimes)

**Comment by aldonetti** [comment]
I know the feeling @glatzert :) and I fully understand your point. 
I'm just saying that at some point we had that in MSDN but there were cases where the users had their global/cookie setting to show e.g. italian articles but wanted to see a specific one in english - to do that, they were adding "/en-us/" in the URL and, because of this mechanism, they were still seeing the italian one. 
Another scenario was that the cookie settings were sometimes "lost" or being overwritten by other parts of the site (sites this big are not always implemented as a single one, but are a composition of different sub-sites all looking similar) so users ended up complaining that they were seeing articles in a language that was different than the one they had selected and there was no way to force the language, not even by forcing the locale in the URL...
That's why I'm suggesting it needs to be thought through well (I still do understand your point :))
thanks!

**Comment by glatzert** [comment]
Thanks for the detailed Explanation, I did not have that in mind.

Regarding that, I'd like to rephrase my proposal:
Provide a way, to fast Switch to the originial (read en-us) version of the article, just as you provide a way to activate and deactivate the hover-translations. Perhaps it would also be viable to create a list of languages, which are present as "fast-language-Switch"?

**Comment by jdanyow** [comment]
maybe something like this?

<img width="889" alt="view-in-english" src="https://user-images.githubusercontent.com/1306747/34580359-ae29505c-f15a-11e7-9b90-6e4686ffe9f1.png">

**Comment by glatzert** [comment]
Something like this would definately make it more comfortable to jump to the original. :)

**Comment by cyrusdargahi** [comment]
Oooh @jdanyow suggestion looks great! I tried to provide the same feedback over Twitter, which didn't go so well :).  But then I remembered that you guys are on GitHub and so I found this issue immediately. 

In my opinion adding the "View in English" button is the best and UX friendly solution without creating a complicated solution where one flag takes precedence over another flag and so on, as of course you guys have already mentioned.

I also sometimes find my self reading parts of the doc in English and then I follow a link in the doc to only find myself in a machine translated page in the Swedish language (my locale), which then disrupts my flow.

Just wanted to also highlight that putting the "language switch" option at bottom left part of the screen might not be the most user friendly solution for the following reason:

1.  it is not easy to find
2.  the user has to scroll the entire page to find it
3.  it is pretty disruptive in the navigation/work flow as it takes the user to a completely different view and then again to the selected language.

This might be a bit off topic, but I thought worth mentioning as it might not always be clear in your developer work flows, if you guys are "testing"/interacting with the site and mainly have your locale set to English.
